src/browser/cached_resource_fetcher.ts

```diff
@@ -120,8 +120,8 @@ function generateHash(str: string): string {
 function download(fileUrl: string, filePath: string, callback: (error: null|Error, filePath: string) => any): void {
     const fetcher = new resourceFetcher('file');
 
-    fetcher.once('fetch-complete', (event: string, status: string) => {
-        if (status === 'success') {
+    fetcher.once('fetch-complete', (event: string, status: string, data: string, headers: any) => {
+        if (status === 'success' && /^2/.test(headers.statusCode)) {
```

Comment:
2XX and 3XX should be considered successful status codes.

Also move this into a descriptive variable/constant.